components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -75,8 +103,35 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
       this.setState({ error });
     }
 
+    // we're coming back from a workspace cluster/IDE URL where we expected a workspace to be running but it wasn't because either:
+    //  - this is a (very) old tab and the workspace already timed out
+    //  - due to a start error our workspace terminated very quickly between:
+    //    a) us being redirected to that IDEUrl (based on the first ws-manager update) and
+    //    b) our requests being validated by ws-proxy
+    // we break this potentially DDOS cycle by not calling "startWorkspace" automatically but offering a "Open workspace" button instead.
+    const params = this.props.parameters;
+    if (!!params?.not_found) {
```

Comment:
would be nice to name the parameter after what this component actually does (i.e. `noAutoStart`) instead of after one situation in which we would want to have that behavior. Then we also don't need the lengthy comment above.

Reply:
Thx for sharing.

I personally prefer:
 a) to decouple ws-proxy from dashboard (by not controlling behavior)
 b) the lengthy comment: This code is already hopelessly overloaded, and explanations including contexts enable others to follow along without needing to debug every path, again.

I agree that we could map `not_found` => `dontAutostart` outside, though, and make it part of the props. :+1:

Reply:
Done.